src/policy/policy.cpp

```diff
@@ -186,20 +187,36 @@ bool AreInputsStandard(const CTransaction& tx, const CCoinsViewCache& mapInputs)
         std::vector<std::vector<unsigned char> > vSolutions;
         TxoutType whichType = Solver(prev.scriptPubKey, vSolutions);
         if (whichType == TxoutType::NONSTANDARD || whichType == TxoutType::WITNESS_UNKNOWN) {
+            reason = "bad-txns-input-script-nonstandard";
+            debug = strprintf("input %u", i);
+            return false;
+        } else if (whichType == TxoutType::WITNESS_UNKNOWN) {
             // WITNESS_UNKNOWN failures are typically also caught with a policy
             // flag in the script interpreter, but it can be helpful to catch
             // this type of NONSTANDARD transaction earlier in transaction
             // validation.
+            reason = "bad-txns-input-witness-unknown";
+            debug = strprintf("input %u", i);
             return false;
```

Comment:
This if-branch is currently dead code, since the type `WITNESS_UNKNOWN` is already matched in the first if-condition a few lines above (obvioulsy, it was forgotten to be removed there). I.e. we want

```diff
diff --git a/src/policy/policy.cpp b/src/policy/policy.cpp
index 08f41376a..581fdcce3 100644
--- a/src/policy/policy.cpp
+++ b/src/policy/policy.cpp
@@ -186,7 +186,7 @@ std::optional<std::pair<std::string, std::string>> AreInputsStandard(const CTran

         std::vector<std::vector<unsigned char> > vSolutions;
         TxoutType whichType = Solver(prev.scriptPubKey, vSolutions);
-        if (whichType == TxoutType::NONSTANDARD || whichType == TxoutType::WITNESS_UNKNOWN) {
+        if (whichType == TxoutType::NONSTANDARD) {
             return std::make_pair(
                 "bad-txns-input-script-nonstandard",
                 strprintf("input %u", i)

```

To avoid issues like this, should probably also add a test for "bad-txns-input-witness-unknown"?

Reply:
Good point! Updated here: https://github.com/bitcoin/bitcoin/compare/631d3c980090343f2f0d9dd97dac585c7fa94f13..1f3c017cce798bdcee1d719bf160dc07783239f2